torch/csrc/jit/passes/reconstruct_scopes.cpp

```diff
@@ -7,77 +7,163 @@ namespace jit {
 class ReconstructScopesPass {
  public:
   ReconstructScopesPass(const Module& m, Graph& g, std::string p)
-      : root_module_(m), graph_(g), prefix_(std::move(p)){};
+      : root_module_(m),
+        graph_(g),
+        prefix_(std::move(p)),
+        class_types_are_not_unique_(false){};
   void run();
 
  private:
   const Module& root_module_;
   Graph& graph_;
   std::string prefix_;
 
+  // This boolean indicates whether there are two submodules of the same
+  // class type. This issue may occur in q scripted module and make it
+  // difficult to exactly track module information corresponding to each
+  // Node* after inlining the graph. Consider the following example:
```

Comment:
I think we should have kept all the changes to reconstruct scopes in base diff. It makes it easy to revert a particular commit.

Reply:
Makes sense. I think I'll move every change to `reconstruct_scopes.cpp` from the top diff to the base diff.

Reply:
Thanks Tai. I understand this is a bit cumbersome, but it looks clean.